atoma-proxy/src/server/middleware.rs

```diff
@@ -671,10 +668,10 @@ pub(crate) mod auth {
         skip_all,
         fields(endpoint = %endpoint)
     )]
-    pub(crate) async fn authenticate_and_lock_compute_units(
+    pub async fn authenticate_and_lock_compute_units(
         state: &ProxyState,
         headers: &HeaderMap,
-        request_model: impl RequestModel,
+        request_model: impl RequestModel + Send,
```

Comment:
Do we really need to mark it as `Send`?

Reply:
Yes this is actually quite important because a future cannot be sent between threads safely

More info here https://rust-lang.github.io/rust-clippy/master/index.html#future_not_send

Reply:
I know, but we were not doing it so far, otherwise the code would not compile

Reply:
But it is fine, let's leave it as is

Reply:
Good point, it compiles because the concrete implementations of `RequestModel` implement `Send`. 

But I think it's more robust to explicitly mark the requirement for `Send` to catch potential future changes that might introduce non-`Send` types and also provide better error messages. If something non-`Send` is introduced should we add another concrete implementation - at the very least this can't hurt.